src/snowflake/snowpark/session.py

```diff
@@ -1684,8 +1683,10 @@ def range(self, start: int, end: Optional[int] = None, step: int = 1) -> DataFra
         if self.sql_simplifier_enabled:
             df = DataFrame(
                 self,
-                SelectStatement(
-                    from_=SelectSnowflakePlan(range_plan, analyzer=self._analyzer),
+                self._analyzer.create_SelectStatement(
+                    from_=self._analyzer.create_SelectSnowflakePlan(
+                        range_plan, analyzer=self._analyzer
+                    ),
```

Comment:
is this trying to unify mock and non-mock?

what's difference between calling `SelectStatement` and calling `self._analyzer.create_SelectStatement` or non-mocking execution, will there be any side-effect?

(and same question to `create_SelectSnowflakePlan`)

Reply:
The difference lies in `self._analyzer`, `MockAnalyzer` should only create `MockSelectable` and real `Analyzer` should only create real `Selectable`.
Yes this is for unifying codepath and minimizing change, although I don't have a strong preference here.

Reply:
I checked the real `Analyzer`, this change should be good!